src/components/App.tsx

```diff
@@ -163,14 +164,40 @@ function App() {
   const [nodes, setNodes, onNodesChange] = useNodesState([]);
   const [edges, setEdges, onEdgesChange] = useEdgesState([]);
 
+  const edgeUpdateSuccessful = useRef(true);
+
+  const onEdgeUpdateStart = useCallback(() => {
+    edgeUpdateSuccessful.current = false;
+  }, []);
+
+  const onEdgeUpdate = useCallback((oldEdge: Edge<any>, newConnection: Connection) => {
+    if (getFluxNodeParent(nodes, edges, newConnection.target!) !== undefined) {
+      return;
+    }
+
+    edgeUpdateSuccessful.current = true;
+    setEdges((edges) => updateEdge(oldEdge, newConnection, edges));
+  }, [nodes, edges]);
+
+  const onEdgeUpdateEnd = useCallback((_: unknown, edge: Edge<any>) => {
+    if (!edgeUpdateSuccessful.current) {
+      setEdges((edges) => edges.filter((e) => e.id !== edge.id));
+    }
+
+    edgeUpdateSuccessful.current = true;
+  }, []);
+
   const onConnect = (connection: Edge<any> | Connection) => {
```

Comment:
also while we're at it, let's useCallback this?

Reply:
ah well wait we depend on `edges` AND `nodes` here... nvm too much complexity for this PR

can maybe do in a followup https://github.com/transmissions11/flux/issues/31